diff --git a/modules/audiounit/recorder.c b/modules/audiounit/recorder.c
index 663fc3d..ae7ca95 100644
--- a/modules/audiounit/recorder.c
+++ b/modules/audiounit/recorder.c
@@ -113,6 +113,11 @@ int audiounit_recorder_alloc(struct ausrc_st **stp, const struct ausrc *as,
                kAudioHardwarePropertyDefaultInputDevice,
                kAudioObjectPropertyScopeGlobal,
                kAudioObjectPropertyElementMaster };
+        AudioObjectPropertyAddress srateAddress = {
+                kAudioDevicePropertyNominalSampleRate,
+                kAudioObjectPropertyScopeGlobal,
+                kAudioObjectPropertyElementMaster };
+        Float64 sampleRate = 48000.0;
        Float64 hw_srate = 0.0;
        UInt32 hw_size = sizeof(hw_srate);
        OSStatus ret = 0;
@@ -165,6 +170,21 @@ int audiounit_recorder_alloc(struct ausrc_st **stp, const struct ausrc *as,
        if (ret)
                goto out;

+       ret = AudioObjectSetPropertyData(inputDevice,
+                       &srateAddress,
+                       0,
+                       NULL,
+                       sizeof(sampleRate),
+                       &sampleRate);
+       if (!ret) {
+               ret = AudioUnitSetProperty(st->au,
+                               kAudioUnitProperty_SampleRate,
+                               kAudioUnitScope_Input,
+                               0,
+                               &sampleRate,
+                               sizeof(sampleRate));
+       }
+
        ret = AudioUnitSetProperty(st->au,
                        kAudioOutputUnitProperty_CurrentDevice,
                        kAudioUnitScope_Global,
