diff --git a/src/ice/comp.c b/src/ice/comp.c
index 4454dd9..782e62d 100644
--- a/src/ice/comp.c
+++ b/src/ice/comp.c
@@ -203,6 +203,7 @@ void icem_comp_set_selected(struct icem_comp *comp, struct ice_candpair *cp)
                DEBUG_WARNING("{%s.%u} set_selected: invalid state %s\n",
                              comp->icem->name, comp->id,
                              ice_candpair_state2name(cp->state));
+               return;
        }
 
        mem_deref(comp->cp_sel);
