From 9cfc8a683a136bbba6d627d69afb71d96dc17224 Mon Sep 17 00:00:00 2001
From: Sebastian Reimers <info@miete-admin.de>
Date: Sun, 9 Oct 2016 13:41:17 +0200
Subject: [PATCH] add samplerate patch

---
 modules/audiounit/recorder.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/modules/audiounit/recorder.c b/modules/audiounit/recorder.c
index cf6a1af..1dcf601 100644
--- a/modules/audiounit/recorder.c
+++ b/modules/audiounit/recorder.c
@@ -114,6 +114,11 @@ int audiounit_recorder_alloc(struct ausrc_st **stp, const struct ausrc *as,
 		kAudioHardwarePropertyDefaultInputDevice,
 		kAudioObjectPropertyScopeGlobal,
 		kAudioObjectPropertyElementMaster };
+	AudioObjectPropertyAddress srateAddress = {
+		kAudioDevicePropertyNominalSampleRate,
+		kAudioObjectPropertyScopeGlobal,
+		kAudioObjectPropertyElementMaster };
+	Float64 sampleRate = 48000.0;
 #endif
 	Float64 hw_srate = 0.0;
 	UInt32 hw_size = sizeof(hw_srate);
@@ -176,6 +181,21 @@ int audiounit_recorder_alloc(struct ausrc_st **stp, const struct ausrc *as,
 			sizeof(inputDevice));
 	if (ret)
 		goto out;
+
+	ret = AudioObjectSetPropertyData(inputDevice,
+			&srateAddress,
+			0,
+			NULL,
+			sizeof(sampleRate),
+			&sampleRate);
+	if (!ret) {
+		ret = AudioUnitSetProperty(st->au,
+				kAudioUnitProperty_SampleRate,
+				kAudioUnitScope_Input,
+				0,
+				&sampleRate,
+				sizeof(sampleRate));
+	}
 #endif
 
 	fmt.mSampleRate       = prm->srate;
