diff --git a/include/re_net.h b/include/re_net.h
index 04d87f3..eb0aa45 100644
--- a/include/re_net.h
+++ b/include/re_net.h
@@ -40,6 +40,7 @@ struct sa;
 
 /* Net generic */
 int  net_hostaddr(int af, struct sa *ip);
+int  net_dst_source_addr_get(struct sa *dst, struct sa *ip);
 int  net_default_source_addr_get(int af, struct sa *ip);
 int  net_default_gateway_get(int af, struct sa *gw);
 
diff --git a/src/net/net.c b/src/net/net.c
index 0978b1b..4c38eb2 100644
--- a/src/net/net.c
+++ b/src/net/net.c
@@ -17,6 +17,8 @@
 #include <re_mbuf.h>
 #include <re_sa.h>
 #include <re_net.h>
+#include <re_udp.h>
+#include <re_mem.h>
 
 
 #define DEBUG_MODULE "net"
@@ -56,6 +58,43 @@ int net_hostaddr(int af, struct sa *ip)
 }
 
 
+/**
+ * Get the source IP address for a specified destination
+ *
+ * @param dst Destination IP address
+ * @param ip  Returned IP address
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int net_dst_source_addr_get(struct sa *dst, struct sa *ip)
+{
+	int err;
+	struct udp_sock *us;
+
+	if (sa_af(dst) == AF_INET6)
+		err = sa_set_str(ip, "::", 0);
+	else
+		err = sa_set_str(ip, "0.0.0.0", 0);
+
+	if (err)
+		return err;
+
+	err = udp_listen(&us, ip, NULL, NULL);
+	if (err)
+		return err;
+
+	err = udp_connect(us, dst);
+	if (err)
+		goto out;
+
+	err = udp_local_get(us, ip);
+
+out:
+	mem_deref(us);
+	return err;
+}
+
+
 /**
  * Get the default source IP address
  *
@@ -66,10 +105,34 @@ int net_hostaddr(int af, struct sa *ip)
  */
 int net_default_source_addr_get(int af, struct sa *ip)
 {
+	struct sa dst;
+	int err;
+#if !defined(WIN32)
+	char ifname[64] = "";
+#endif
+
+	sa_init(&dst, af);
+
+	if (af == AF_INET6)
+		err = sa_set_str(&dst, "1::1", 53);
+	else
+		err = sa_set_str(&dst, "1.1.1.1", 53);
+
+	err = net_dst_source_addr_get(&dst, ip);
+
+	if (af == AF_INET6) {
+		if (sa_is_linklocal(ip)) {
+			sa_init(ip, af);
+			return 0;
+		}
+	}
+
+	if (!err)
+		return 0;
+
 #if defined(WIN32)
 	return net_hostaddr(af, ip);
 #else
-	char ifname[64] = "";
 
 #ifdef HAVE_ROUTE_LIST
 	/* Get interface with default route */
