diff --git a/src/stream.c b/src/stream.c
index 99f3283..87ac887 100644
--- a/src/stream.c
+++ b/src/stream.c
@@ -470,6 +470,7 @@ int stream_alloc(struct stream **sp, struct list *streaml,
 {
 	struct stream *s;
 	int err;
+	bool rtcp_mux;
 
 	if (!sp || !prm || !cfg || !rtph)
 		return EINVAL;
@@ -478,6 +479,12 @@ int stream_alloc(struct stream **sp, struct list *streaml,
 	if (!s)
 		return ENOMEM;
 
+	if (str_cmp(mnat->id, "turn") == 0)
+		rtcp_mux = false;
+	else
+		rtcp_mux = cfg->rtcp_mux;
+
+
 	MAGIC_INIT(s);
 
 	s->cfg   = *cfg;
@@ -532,7 +539,7 @@ int stream_alloc(struct stream **sp, struct list *streaml,
 				   rtp_sess_ssrc(s->rtp), prm->cname);
 
 	/* RFC 5761 */
-	if (cfg->rtcp_mux &&
+	if (rtcp_mux &&
 	    (offerer || sdp_media_rattr(s->sdp, "rtcp-mux"))) {
 
 		err |= sdp_media_set_lattr(s->sdp, true, "rtcp-mux", NULL);
@@ -545,7 +552,7 @@ int stream_alloc(struct stream **sp, struct list *streaml,
 		s->mnat = mnat;
 		err = mnat->mediah(&s->mns, mnat_sess,
 				   rtp_sock(s->rtp),
-				   cfg->rtcp_mux ? NULL : rtcp_sock(s->rtp),
+				   rtcp_mux ? NULL : rtcp_sock(s->rtp),
 				   s->sdp, mnat_connected_handler, s);
 		if (err)
 			goto out;
