diff -Naur re-0.4.14/mk/re.mk re-0.4.14_mod/mk/re.mk
--- re-0.4.14/mk/re.mk	2015-06-29 08:51:05.000000000 +0200
+++ re-0.4.14_mod/mk/re.mk	2015-12-25 08:52:03.678335804 +0100
@@ -433,7 +433,7 @@
 
 ifneq ($(USE_OPENSSL),)
 CFLAGS  += -DUSE_OPENSSL -DUSE_TLS
-LIBS    += -lssl -lcrypto
+LIBS    += -lssl -lws2_32 -lgdi32 -lcrypt32 -lcrypto -lws2_32 -lgdi32 -lcrypt32
 USE_TLS := yes
 
 USE_OPENSSL_DTLS := $(shell [ -f $(SYSROOT)/include/openssl/dtls1.h ] || \
diff -Naur re-0.4.14/src/tls/openssl/tls.c re-0.4.14_mod/src/tls/openssl/tls.c
--- re-0.4.14/src/tls/openssl/tls.c	2015-04-10 13:02:57.000000000 +0200
+++ re-0.4.14_mod/src/tls/openssl/tls.c	2015-12-23 18:22:56.193450682 +0100
@@ -22,7 +22,6 @@
 #include <re_tls.h>
 #include "tls.h"
 
-
 #define DEBUG_MODULE "tls"
 #define DEBUG_LEVEL 5
 #include <re_dbg.h>
@@ -204,6 +203,7 @@
  *
  * @return 0 if success, otherwise errorcode
  */
+#if 0
 int tls_set_selfsigned(struct tls *tls, const char *cn)
 {
 	X509_NAME *subj = NULL;
@@ -305,7 +305,7 @@
 
 	return err;
 }
-
+#endif
 
 /**
  * Set the certificate and private key on a TLS context
